src/Files.App.Controls/ThemedIcon/ThemedIcon.cs

```diff
@@ -0,0 +1,387 @@
+                {
+                    layer.Width = newWidth;
+                    layer.Height = newHeight;
+                }
+            }
+        }
+
+        private void EnabledChanged(bool value)
+        {
+            // Handles the IsEnabled property change
+
+            _isEnabled = value;
+
+            UpdateVisualStates();
+        }
+
+        private void ThemeSettings_OnHighContrastChanged(object sender, bool e)
+        {
+            HighContrastChanged(e);
+        }
+
+        private void HighContrastChanged(bool value)
+        {
+            // handles HighContrast property change
+
+            _isHighContrast = value;
+
+            UpdateVisualStates();
+        }
+
+        private void InitialIconStateValues()
+        {
+            _isEnabled = IsEnabled;
+            _isToggled = IsToggled;
+            _isHighContrast = IsHighContrast;
+        }
+
+        private void UpdateIconStates()
+        {
+            ToggleChanged(_isToggled);
+            EnabledChanged(_isEnabled);
+            HighContrastChanged(_isHighContrast);
+        }
+
+        private void UpdateVisualStates()
+        {
+            // Updates all Icon Visual States.
+
+            UpdateIconTypeStates();
+            UpdateIconColorTypeStates();
+        }
+
+        private void UpdateIconTypeStates()
+        {
+            // Handles changes to the IconType and setting the correct Visual States.
+
+            // Handles the two IconType states, based on the ThemedIcon.IconType value
+            // as well as states derived from owner controls, and other properties
+
+            // We first check for isToggled and isFilled icon types and states
+            // Then we check for Contrast and Disabled states, to replace Layered with Outline and set EnabledStates
+            // Finally we assigned Filled and Layered states, and default otherwise to Outline
+
+            if (_isToggled is true || IsToggled is true || _isFilled is true || IsFilled is true)
+            {
+                VisualStateManager.GoToState(this, FilledTypeStateName, true);
+                return;
+            }
+            else if (_isHighContrast is true || IsHighContrast is true || _isEnabled is false || IsEnabled is false)
+            {
+                VisualStateManager.GoToState(
+                    this,
+                    IconType switch
+                    {
+                        ThemedIconTypes.Layered => OutlineTypeStateName,
+                        _ => OutlineTypeStateName,
+                    },
+                    true);
+
+                VisualStateManager.GoToState(this, DisabledStateName, true);
+            }
+            else
+            {
+                VisualStateManager.GoToState(
+                    this,
+                    IconType switch
+                    {
+                        ThemedIconTypes.Layered => LayeredTypeStateName,
+                        _ => OutlineTypeStateName,
+                    },
```

Comment:
We originally had 3 states handled with this code, I removed the Filled ThemedIconType, and made it a Bool property.  I simply removed one of the lines, and left the code as is.  @0x5bfa re-wrote the nested if statements I wrote originally.

Reply:
@0x5bfa If you can tell me what the Ternary version of this code should be, I can do it.  As it stands, I don't understand how I would do it lol

Reply:
IconType is ThemedIconTypes.Layered ? LayeredTypeStateName : OutlineTypeStateName


```cs
if (condition)
  return consequence;
else
  return alternative;
```

to

```cs 
condition ? consequence : alternative;
```

This can be used not only in return state but everywhere in any statement.